chain/consensus/mir/manager.go

```diff
@@ -182,23 +181,26 @@ func NewManager(ctx context.Context, addr address.Address, api v1api.FullNode) (
 
 // Start starts the manager.
 func (m *Manager) Start(ctx context.Context) chan error {
-	log.Info("Mir manager starting")
+	log.Infof("Mir manager %s starting", m.MirID)
 
 	errChan := make(chan error, 1)
-	managerCtx, managerCancel := context.WithCancel(ctx)
+	managerCtx, managerCancel := context.WithCancel(context.Background())
 
 	go func() {
 		select {
 		case <-ctx.Done():
-			log.Debugf("Mir manager: context closed")
+			log.Infof("Mir manager %s: context closed", m.MirID)
 			m.Stop()
```

Comment:
Is the manager supposed to stop only when the context given to it from outside is canceled, or should it also stop when the Mir node stops by itself for some reason (e.g. due to an error)?
Currently, if the Mir node encounters an error, the Manager is not stopped.
Moreover, if the Mir node stops by itself without an error, this whole goroutine will keep running until the external context is canceled.
Is that intended?

Reply:
1. Case 1. If the Mir node encounters an error then we send the error to `errChan` and call `managerCancel()` that will stop the manager as a goroutine but it will not call the `Stop` function. `Stop` doesn't stop the manager it stops the Mir node. The question: if the Mir node has encountered an error does make sense to stop its components gracefully?
2. Case 2. `Mir node stops by itself without an error`. According to the comments it `Stops and returns when ctx is canceled.` So it can't just finish all work and stop. Right? It stops via context only and then the special error is returned.

Reply:
Then I don't quite understand it... In the code it looks like `m` is the Manager and the code calls `m.Stop()`. Doesn't that stop the Manager?

Reply:
> if the Mir node has encountered an error does make sense to stop its components gracefully?

Yes. The Mir Node does not really have a reason (or even a way) of stopping other modules like the WAL or Net on error. So in fact, they still need to be stopped from the outside event if the Node fails. In a way, the modules can be seen as components of the Node, but one can also see them as independent components that only interact using the node (after all they are also started separately from the Node). So it make sense to also stop them (preferably gracefully) separately from the Node (even if it fails).

Reply:
fixed